userparameters/ZoL_with_sudo.conf

```diff
@@ -1,9 +1,19 @@
 # ZFS discovery and configuration
 # original template from pbergbolt (source = https://www.zabbix.com/forum/showthread.php?t=43347), modified by Slash <slash@aceslash.net>
 
-
 # pool discovery
 UserParameter=zfs.pool.discovery,/usr/bin/sudo /sbin/zpool list -H -o name | sed -e '$ ! s/\(.*\)/{"{#POOLNAME}":"\1"},/' | sed -e '$  s/\(.*\)/{"{#POOLNAME}":"\1"}]}/' | sed -e '1  s/\(.*\)/{ \"data\":[\1/'
+
+# pool IOPS read
+UserParameter=zfs.zpool.iostat.reads[*],/usr/bin/sudo /sbin/zpool iostat $1 -H -p 1 2 | sed 's/\t/-/g' | cut -d '-' -f 4  | tail -n 1
```

Comment:
Executing the same command multiple times just to extract different fields from the output is both extremely inefficient and has potential inconsistency problems.

It would be much better to run the command once and then use [dependent items](https://www.youtube.com/watch?v=1omyJXYqsiI&t=17m4s) to just parse the output multiple times instead of running the command multiple times.